File: 2021/09/09/main.go
package main

func orderOfLargestPlusSign(n int, mines [][]int) int {
	k := n/2 + n%2

	if len(mines) == 0 {
		return k
	}

	isMined := [500][500]bool{}
	for _, m := range mines {
		isMined[m[0]][m[1]] = true
	}

	isGoodEdge := func(xBegin, xEnd, yBegin, yEnd int) bool {
		for i := xBegin; i <= xEnd; i++ {
			for j := yBegin; j <= yEnd; j++ {
				if isMined[i][j] {
					return false
				}
			}
		}
		return true
	}

	for ; k > 0; k-- {
		for i := k - 1; i <= n-k; i++ {
			for j := k - 1; j <= n-k; j++ {
				if !isMined[i][j] &&
					isGoodEdge(i-k+1, i-1, j, j) &&
					isGoodEdge(i+1, i+k-1, j, j) &&
					isGoodEdge(i, i, j-k+1, j-1) &&
					isGoodEdge(i, i, j+1, j+k-1) {
					return k
				}
			}
		}
	}

	return 0
}

File: 2021/09/09/main.py
class Solution:
    def orderOfLargestPlusSign(self, n: int, mines: List[List[int]]) -> int:
        dp, g, m = [[0]*n for i in range(n)], {tuple(m) for m in mines}, 0
        for i in range(n):
            c = 0
            for j in range(n):
                c = 0 if (i, j) in g else c + 1
                dp[i][j] = c
            c = 0
            for j in range(n-1, -1, -1):
                c = 0 if (i, j) in g else c + 1
                dp[i][j] = min(dp[i][j], c)
        for j in range(n):
            c = 0
            for i in range(n):
                c = 0 if (i, j) in g else c + 1
                dp[i][j] = min(dp[i][j], c)
            c = 0
            for i in range(n-1, -1, -1):
                c = 0 if (i, j) in g else c + 1
                m = max(m, min(dp[i][j], c))
        return m